core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -77,6 +79,7 @@
 
   // Single-write, multi-read lock used to create the first match
   private final ReentrantReadWriteLock lock;
+  private boolean oldFireTick = true;
```

Comment:
this looks like a pretty odd place for this.

Consider having GameRulesMatchModule have logic for getting all gamerules in the level.dat, then ammending the ones defined in the XML ontop of them, and being able to provide them via getter, that way you can simply use them anywhere without having to save random booleans all across the codebase